app/controllers/items/detail.js

```diff
@@ -154,6 +154,15 @@ export default GoodcityController.extend(
       }
     }),
 
+    canApplyDefaultValuation: Ember.computed("model.valueHkDollar", function() {
+      const valueHkDollar = +this.get("model.valueHkDollar");
+      let defaultValue = +this.get("defaultValueHkDollar");
+      defaultValue = !!defaultValue
+        ? defaultValue
+        : this.set("defaultValueHkDollar", valueHkDollar);
```

Comment:
doing a `set` in a computed property is always a code smell. Better keep those safe and readonly, state should be modified at initialisation or through user actions

Reply:
Thanks @patrixr , This was a ToDo and I had missed it. Thanks for pointing it out